src/autopas/tuning/tuningStrategy/RandomSearch.cpp

```diff
@@ -15,21 +15,27 @@ autopas::RandomSearch::RandomSearch(size_t maxEvidence, unsigned long seed)
     : _rng(seed), _maxEvidence(maxEvidence), _numEvidenceCollected(0) {}
 
 void autopas::RandomSearch::optimizeSuggestions(std::vector<Configuration> &configQueue,
-                                                const EvidenceCollection &evidenceCollection) {
+                                                const EvidenceCollection &evidenceCollection,
+                                                std::optional<std::reference_wrapper<bool>> intentionalConfigWipe) {
   // Check if enough evidence was collected.
   if (_numEvidenceCollected == _maxEvidence) {
     // Remove everything from the queue and only insert the optimum.
     configQueue.clear();
+    if (intentionalConfigWipe) {
+      intentionalConfigWipe->get() = true;
+    }
   } else {
     // Swap a randomly selected configuration to the end.
     std::uniform_int_distribution<std::mt19937::result_type> distribution(0, configQueue.size() - 1);
     const auto selectedIndex = distribution(_rng);
     std::swap(configQueue.back(), configQueue[selectedIndex]);
+    ++_numEvidenceCollected;
   }
```

Comment:
I noticed that `_numEvidenceCollected` is never incremented. Accordingly, you never get into the case that `_numEvidenceCollected == _maxEvidence` and the config is wiped. I have now changed this and `_numEvidenceCollected` is incremented. Hope this is correct.

Reply:
Good catch that is indeed a mistake!

The increment should, however, happen in `addEvidence()` for which there is no override yet so please add that and move it there.

Reply:
done in [976b882](https://github.com/AutoPas/AutoPas/pull/936/commits/976b88217759053e644121139c2901b725f69ffc)